### PR TITLE
Strip trailing slashes to avoid 404 from Home Assistant's API endpoint

### PR DESCRIPTION
**Pull request by lassebm**
⚠️⚠️⚠️ Since we do not accept all types of pull requests and do not want to waste your time. Please be sure that you have read pull request rules:
https://github.com/louislam/uptime-kuma/blob/master/CONTRIBUTING.md#can-i-create-a-pull-request-for-uptime-kuma

Tick the checkbox if you understand [x]: 
- [x] I have read and understand the pull request rules.

# Description

Strip trailing slashes to avoid 404 from Home Assistant's API endpoint.

## Type of change

Please delete any options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I ran ESLint and other linters for modified files
- [x] I have performed a self-review of my own code and tested it
- [ ] I have commented my code, particularly in hard-to-understand areas
  (including JSDoc for methods)
- [x] My changes generate no new warnings
- [ ] My code needed automated testing. I have added them (this is optional task)

## Screenshots (if any)

Please do not use any external image service. Instead, just paste in or drag and drop the image here, and it will be uploaded automatically.

### CODE REVIEW

File: server/notification-providers/home-assistant.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ class HomeAssistant extends NotificationProvider {
 
         try {
             await axios.post(
-                `${notification.homeAssistantUrl}/api/services/notify/${notificationService}`,
+                `${notification.homeAssistantUrl.replace(/\/*$/, "")}/api/services/notify/${notificationService}`,
```
Comment (louislam) [inline server/notification-providers/home-assistant.js:14]: `.trim()` should be better?

Comment (lassebm) [reply server/notification-providers/home-assistant.js:14]: `.trim()` won't strip trailing slashes, just whitespace. Trimming before stripping would be a good idea though, I'll add.

Comment (louislam) [reply server/notification-providers/home-assistant.js]: Understood, I wrongly read the title.

**Comment by ThomDietrich** [comment]
Hey @lassebm,
just out of interest. How did you define your Home Assistant check? Mine is configured this way and no fix via a PR was needed. Do you have a better check I might want to adapt?

image

**Comment by lassebm** [comment]
@ThomDietrich If you had a trailing slash in the URL before the PR, HA would 404 due to double slashes in the constructed URL.

**Comment by ThomDietrich** [comment]
ah ^^ okay that makes sense. Thanks for your PR :)

**Pull request merged**